linkerd/meshtls/boring/src/creds/verify.rs

```diff
@@ -0,0 +1,21 @@
+use boring::x509::X509;
+use linkerd_dns_name as dns;
+use linkerd_identity as id;
+use std::io;
+
+pub(crate) fn verify_id(cert: &X509, id: &id::Id) -> io::Result<()> {
```

Comment:
i don't love that this returns an `io::Result`, since the only errors we return here are not, actually, IO errors. should we maybe have a new error type for `verify_id`?

Reply:
i believe all of these errors need to make it to io::Error given where this occurs in the accept path.

Reply:
ahh, ugh, nevermind. this is fine, i guess!

Reply:
it just seems unfortunate for these to be turned into "IO errors" in the error metrics...